scalding-parquet/src/main/scala/com/twitter/scalding/parquet/tuple/TypedParquet.scala

```diff
@@ -0,0 +1,110 @@
+package com.twitter.scalding.parquet.tuple
+
+import _root_.parquet.filter2.predicate.FilterPredicate
+import cascading.scheme.Scheme
+import com.twitter.scalding._
+import com.twitter.scalding.parquet.HasFilterPredicate
+import com.twitter.scalding.parquet.tuple.scheme.{ ParquetWriteSupport, ParquetReadSupport, TypedParquetTupleScheme }
+
+import scala.reflect.ClassTag
+
+/**
+ * Typed parquet tuple
+ * @author Jian Tang
+ */
+object TypedParquet {
+  /**
+   * Create readable typed parquet source.
+   * Here is an example:
+   *
+   *  case class SampleClassB(string: String, int: Int, double: Option[Double], a: SampleClassA)
+   *
+   *  class ReadSupport extends ParquetReadSupport[SampleClassB] {
+   *    import com.twitter.scalding.parquet.tuple.macros.Macros._
+   *    override val tupleConverter: ParquetTupleConverter = caseClassParquetTupleConverter[SampleClassB]
+   *    override val rootSchema: String = caseClassParquetSchema[SampleClassB]
+   *  }
```

Comment:
You can call a macro inside a macro, so you could actually have the `ReadSupport` types generated as well.